src/components/FallbackSignal.tsx

```diff
@@ -0,0 +1,63 @@
+import { FC } from "react"
+import { shallow } from "zustand/shallow"
+
+import clsxm from "@/lib/clsxm"
+
+import Tooltip from "@/components/Tooltip"
+
+import { useQueryStatusStore } from "@/store"
+
+type FallbackSignalProps = {
+  className?: string
+  showTooltip?: boolean
+  showLabels?: boolean
+}
+
+const FallbackSignal: FC<FallbackSignalProps> = ({
+  className,
+  showLabels = true,
+}) => {
+  const [failedQueries] = useQueryStatusStore(
+    (state) => [state.queryStatus],
+    shallow
+  )
+  const failedQueriesCount = failedQueries.size
+  const label =
+    failedQueriesCount >= 1 && failedQueriesCount <= 3
```

Comment:
Maybe worth pulling these out into variables since you're using them 3 times in here. 

```ts
const isMinorInterruption = failedQueriesCount >= 1 && failedQueriesCount <= 3
const isCriticalInterruption = failedQueriesCount >= 5
```